src/config.rs

```diff
@@ -0,0 +1,72 @@
+use crate::weight::{WeightScale, ZeroWeightScale};
+use std::collections::hash_map::RandomState;
+use std::hash::BuildHasher;
+use std::marker::PhantomData;
+use std::num::NonZeroUsize;
+
+/// A configuration structure used to create an LRU cache.
+pub struct CLruCacheConfig<K, V, S = RandomState, W = ZeroWeightScale> {
+    pub(crate) capacity: NonZeroUsize,
+    pub(crate) hash_builder: S,
+    pub(crate) reserve: Option<usize>,
+    pub(crate) scale: W,
+    _marker: PhantomData<(K, V)>,
+}
+
+impl<K, V> CLruCacheConfig<K, V> {
+    /// Creates a new configuration that will create an LRU cache
+    /// that will hold at most `capacity` elements and default parameters.
+    pub fn new(capacity: NonZeroUsize) -> Self {
```

Comment:
There are multiple places in the code base where having to handle the "no capacity" case makes the implementation more difficult that it needs to be. For the situations you mentioned would using an `Option<LruCache>` be workable?

```Rust
let cache = NonZeroUsize::new(cap).map(|cap| CLruCache::new(cap));
```

Reply:
Especially `put_or_modify` and `try_put_or_modify` returns a mutable reference to the inserted value which means that in order to relax the capacity constraint we would either need to
1. return an `Option<&mut V>` to be able to handle the case where a key cannot be inserted
2. accept to panic in those functions, which I am not in favor of because, well, no one wants to encounter hidden panics in production

Reply:
One (crazy) idea could be to add another generic type for the capacity, something like `C: Into<usize>` and only implement `(try_)put_modify` for the `NonZeroUsize` type capacity case.

Reply:
> would using an `Option<LruCache>` be workable?

Yes. I wouldn't like to do something like that just for the benchmarks, though. This is not a ig deal. Adding a `disable` flag or similar sounds more reasonable.